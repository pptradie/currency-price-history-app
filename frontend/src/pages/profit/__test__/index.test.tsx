import * as React from "react";
import * as renderer from "react-test-renderer";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import MaxProfitPerDayList from "../index";
import configureMockStore from "redux-mock-store";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe("MaxProfitPerDayList component", () => {
	let defaultProps = {
		maxProfitPerDayState: {
			maxProfitPerDay: {
				ETC: {
					currency: "ETC",
					date: "20180507",
					buy: {
						time: "0930",
						price: 14.32
					},
					sell: {
						time: "1245",
						price: 15.03
					},
					max_profit: 0.71
				}
			}
		}
	};

	it("renders currency profit page", () => {
		const store = mockStore(defaultProps);

		const renderedMaxProfitPerDayList = renderer
			.create(
				<Provider store={store}>
					<MaxProfitPerDayList currency="ETC" />
				</Provider>
			)
			.toJSON();
		expect(renderedMaxProfitPerDayList).toMatchSnapshot();
	});
});
