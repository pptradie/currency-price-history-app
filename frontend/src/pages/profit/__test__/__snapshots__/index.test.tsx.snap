// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxProfitPerDayList component renders currency profit page 1`] = `
<tr
  className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-1"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    20180507
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    ETC
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0930
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    $ 
    14.32
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    1245
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    $ 
    15.03
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    $ 
    0.71
  </td>
</tr>
`;
